### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by VLeads**
Added authentication.

### CODE REVIEW

File: src/context/video-context.js [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
   );
 
-  const value = { videos, videosDispatch };
+  const getSingleVideo = async (id, setSingleVideo) => {
+    try {
+      const response = await getVideoApi(id);
+      if (response.status === 200) {
+        setSingleVideo(response.data.video);
+      }
+      console.log("watch", response.data);
```
Comment (ApurvaSawant11) [inline src/context/video-context.js:26]: Remove this and  above comment if not needed

File: src/pages/watch/watch.css [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+.video_avatar img {
+    width: 45px;
+    height: 45px;
+    object-fit: cover;
+    overflow: hidden;
+    border-radius: 50%;
+    margin-right: 15px
+}
+
+.channel_name {}
```
Comment (ApurvaSawant11) [inline src/pages/watch/watch.css:39]: Remove this and on line 1 `.video_page_container {}`

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me!

Just add `Login as a Guest` option.

File: src/components/header/Header.jsx [hunk 3]
```diff
@@ -1,30 +1,62 @@
...
+  const logoutHandler = () => {
+    setShowToast(true);
+    toastDispatch({
+      type: ACTION_TYPE_SUCCESS,
+      payload: `✅ Successfully logged out `,
+    });
+    setTimeout(() => {
+      setShowToast(false);
+    }, 2500);
+    localStorage.removeItem("token");
```
Comment (abhisheksahani-nova) [inline src/components/header/Header.jsx:25]: Good that you are handling logout also

**Review by abhisheksahani-nova — APPROVED** [review]
Great work with functions extractions , nice UI and clean code , look into apurva' suggestions else good to merge

**Pull request merged**